### PR TITLE
feat(ColorPicker): new component

### LINKED ISSUE #377: [Feat]: New ColorPicker component
`Description`: Color Picker allows users to select and apply colors within an application or website. It is widely used in design tools, customization features, and any interface where users need to choose colors, such as for text, backgrounds, or graphical elements.

**_Requirements:_**

_Component Implementation:_

1. Implement the Color Picker component in TypeScript.
2. Utilize SCSS with BEM methodology for styling.
3. Integrate design tokens from @geneui/tokens for consistent colors, font sizes, spacing, and button styles.
4. Support configuration options such as Hex, rgb, rgba selections, common used colors.
5. Ensure proper accessibility, including keyboard navigation and ARIA attributes.

_Unit Tests:_

1. Write unit tests using Jest and React Testing Library.
2. Ensure test coverage for all props, including rendering of different configurations (rgb, hex, rgba selections).
3. Test for edge cases.

_Storybook Stories:_

1. Create a Storybook story for the Color picker component.
2. Showcase different configurations (e.g., rgb, hex, rgba color selection).
3. Create controls for the props so users can interact with the Time picker component in Storybook.
4. Document all props of the public interfaces.

Checklist:

- [ ] Component is implemented in TypeScript.
- [ ] Styling follows SCSS using BEM methodology.
- [ ] Design tokens from @geneui/tokens are used for colors, font sizes, spacing, and button styles.
- [ ] Unit tests written and cover different states and edge cases.
- [ ] Storybook stories created with examples of all props and configuration options.
- [ ] Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by alberthovhannisyan1**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=1346 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-dzwarikzhd.chromatic.com/

### CODE REVIEW

File: src/components/molecules/ColorPicker/ColorPicker.scss [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    &__wrapper {
+        box-sizing: content-box;
+        display: flex;
+        flex-direction: column;
+        gap: var(--guit-ref-spacing-medium);
+        width: 28.8rem;
+        padding: var(--guit-ref-spacing-large);
+
+        .react-colorful {
```
Comment (GaroGabrielyan) [inline src/components/molecules/ColorPicker/ColorPicker.scss]: This approach is unacceptable; we should not be depended from third part library classname.

File: src/components/molecules/ColorPicker/ColorPicker.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+export default meta;
+
+type Story = StoryObj<IColorPickerProps>;
+
+export const Default: Story = {};
+
+export const WithRecentColors: Story = {
+    render: (props) => {
+        return (
+            <div style={{ width: "100%" }}>
```
Comment (GaroGabrielyan) [inline src/components/molecules/ColorPicker/ColorPicker.stories.tsx]: Please remove unnecessary `<div>`
```suggestion
           return <ColorPicker {...props} />;
```

File: src/components/molecules/ColorPicker/ColorPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+
+    useClickOutside(() => {
+        if (!isOpenControlled) {
+            setIsOpen(false);
+            return;
+        }
+        onOutsideClick?.();
+    }, [popoverRef.current.floatingElement, popoverRef.current.referenceElement]);
+
+    const ColorSquareIcon = useCallback(
```
Comment (GaroGabrielyan) [inline src/components/molecules/ColorPicker/ColorPicker.tsx]: This clickable item should be a `<button>` for proper semantics and accessibility, rather than a generic clickable element. Also, `useCallback` is unnecessary here since the callback is not expensive and does not provide a clear optimization benefit. Finally, this code would be more readable if rgba is destructured (e.g., `const { r, g, b, a } = rgba`) before building the rgba string/style.

File: src/components/molecules/ColorPicker/ColorPicker.tsx [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+        if (!isOpenControlled || open === undefined) return;
+        setIsOpen(open);
+    }, [open, isOpenControlled]);
+
+    useEffect(() => {
+        setFormatState(format);
+    }, [format]);
+
+    useClickOutside(() => {
+        if (!isOpenControlled) {
```
Comment (GaroGabrielyan) [inline src/components/molecules/ColorPicker/ColorPicker.tsx:318]: The same controlled-state guard is duplicated in multiple places. Please extract it into a single handler (e.g., `setOpenState(nextState: boolean)`) and reuse it instead of repeating the conditional logic. This will reduce duplication and make future changes safer.

File: src/components/molecules/ColorPicker/ColorPicker.tsx [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+            <Popover
+                onClose={() => !isOpenControlled && setIsOpen(false)}
+                withArrow={false}
+                ref={popoverRef}
+                position="bottom-left"
+                open={isOpen}
+                setProps={setPropsForPopover}
+            >
+                <PopoverBody withPadding={false}>
+                    <div className="colorPicker__wrapper">
```
Comment (GaroGabrielyan) [inline src/components/molecules/ColorPicker/ColorPicker.tsx]: The extra wrapper `<div className="colorPicker__wrapper">` looks unnecessary. We can likely apply this class directly to PopoverBody and keep the same behavior, while removing redundant DOM nesting.
`<PopoverBody withPadding={false} className="colorPicker__wrapper">`

File: src/components/molecules/ColorPicker/ColorPicker.tsx [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+                const next = updater(prev);
+                emitChange(next);
+                setLocalHex(rgbToHex(next));
+                return next;
+            });
+        },
+        [emitChange]
+    );
+
+    const handlePickerChange = useCallback(
```
Comment (GaroGabrielyan) [inline src/components/molecules/ColorPicker/ColorPicker.tsx:183]: This seems like an overuse of useCallback. handlePickerChange is mostly local logic, and memoizing it here adds complexity without clear performance benefit. Unless this callback is required for referential stability in a memoized child, consider using a regular function instead.

File: src/components/molecules/ColorPicker/ColorPicker.tsx [hunk 7]
```diff
@@ -0,0 +1,380 @@
...
+    colorPickerProps,
+    onChange,
+    open,
+    format = "hex",
+    onOutsideClick
+}) => {
+    const isColorControlled = value !== undefined;
+    const isOpenControlled = open !== undefined;
+
+    const [isOpen, setIsOpen] = useState<boolean>(open ?? false);
```
Comment (GaroGabrielyan) [inline src/components/molecules/ColorPicker/ColorPicker.tsx]: No need to explicitly annotate primitive state types here. TypeScript can infer this from the initial value, so useState(open ?? false) is cleaner and equivalent.
This can be simplified from `open ?? false` to `!!open` . The current expression is valid, but the simplified form is more concise.

File: src/components/molecules/ColorPicker/ColorPicker.tsx [hunk 8]
```diff
@@ -0,0 +1,380 @@
...
+    const popoverRef = useRef<IPopoverRef>({
+        floatingElement: { current: null },
+        referenceElement: { current: null }
+    });
+
+    const [rgba, setRgba] = useState<RGBA>(() => {
+        const initialHex = value ?? defaultColor;
+        const rgb = initialHex ? hexToRgb(initialHex) : null;
+
+        return rgb ? { ...rgb, a: alphaValue / 100 } : { ...DEFAULT_RGBA };
```
Comment (GaroGabrielyan) [inline src/components/molecules/ColorPicker/ColorPicker.tsx]: There are several hardcoded `100` values here, which read as magic numbers. Please extract them into well-named constants (e.g., `PERCENT_MAX` or `ALPHA_SCALE_MAX`) so the intent is clear and easier to maintain.

**Review by GaroGabrielyan — CHANGES_REQUESTED** [review]

**Review by GaroGabrielyan — APPROVED** [review]
